src/Menu/menu/sizeModifier.ts

```diff
@@ -0,0 +1,292 @@
+/* eslint-disable @typescript-eslint/ban-ts-ignore */
+// import { Instance, ReferenceElement } from "tippy.js";
+import computeAutoPlacement from "@popperjs/core/lib/utils/computeAutoPlacement";
+import detectOverflow from "@popperjs/core/lib/utils/detectOverflow";
+import getOppositePlacement from "@popperjs/core/lib/utils/getOppositePlacement";
+import getOppositeVariationPlacement from "@popperjs/core/lib/utils/getOppositeVariationPlacement";
+import getVariation from "@popperjs/core/lib/utils/getVariation";
+import {
+  BasePlacement,
+  Boundary,
+  Modifier,
+  ModifierArguments,
+  Padding,
+  Placement,
+  RootBoundary,
+  start,
+} from "@popperjs/core";
+
+interface Options {
+  allowedAutoPlacements: Array<Placement>;
+  altBoundary: boolean;
+  boundary: Boundary;
+  fallbackPlacements: Array<Placement>;
+  flipVariations: boolean;
+  padding: Padding;
+  rootBoundary: RootBoundary;
+}
+
+/**
+ * Get the base of a `Placement`, meaning strip off `-start` and `-end`
+ *
+ * This will also work with `auto`, which is why this is modified from the
+ * internal popper version
+ */
+function getBasePlacement(
+  placement: Placement | "auto"
+): BasePlacement | "auto" {
+  return (placement.split("-")[0] as any) as BasePlacement | "auto";
+}
+
+function getExpandedFallbackPlacements(placement: Placement): Array<Placement> {
+  if (getBasePlacement(placement) === "auto") {
+    return [];
+  }
+
+  const oppositePlacement = getOppositePlacement(placement);
+
+  return [
+    getOppositeVariationPlacement(placement),
+    oppositePlacement,
+    getOppositeVariationPlacement(oppositePlacement),
+  ];
+}
+
+interface OverflowDict {
+  placement: Placement;
+  overflow: ReturnType<typeof detectOverflow>;
+  mainVariationSide: BasePlacement;
+  altVariationSide: BasePlacement;
+  basePlacement: ReturnType<typeof getBasePlacement>;
+}
+
+/**
+ * Given a list of `OverflowDict` and the `state`, find the `OverflowDict` that
+ * should be displayed. Returns the first placement that will fit without
+ * scrolling, and if there are none, then the placement that will require the
+ * _leaset_ amount of scrolling.
+ */
+function findMinimumOverflowPlacement({
+  overflowDicts,
+  state,
+}: {
+  overflowDicts: readonly OverflowDict[];
+  state: ModifierArguments<Options>["state"];
+}) {
+  const overflowsWithOverflowMap = overflowDicts.map((overflowDict) => {
+    const { y } = state.modifiersData.preventOverflow || { y: 0 };
+    const { height } = state.rects.popper;
+    const [basePlacement] = overflowDict.placement.split("-");
+
+    const heightProp = basePlacement === "top" ? "top" : "bottom";
```

Comment:
I don't quite understand this variable name

Since this seems to only be used to reference overflow, how about
```ts
const verticalOverflow = overflowDict.overflow[basePlacement === "top" ? "top" : "bottom"]
```

Reply:
I [stole](https://github.com/atomiks/popper.js/blob/59428b21e4081a80da44c6641a9688e6a862cb7a/src/modifiers/maxSize.js#L17) the code here so I don't know why it's named the way it is. I'm happy to change it to `verticalOverflow`.